vyper/codegen/stmt.py

```diff
@@ -205,24 +207,23 @@ def _get_last(ir):
             ["mstore", buf - 32, 0x20],
             ["revert", buf - 36, ["add", 4 + 32 + 32, ["ceil32", _runtime_length]]],
         ]
-
         if test_expr is not None:
             ir_node = ["if", ["iszero", test_expr], revert_seq]
         else:
             ir_node = revert_seq
-
         return IRnode.from_list(ir_node, error_msg="user revert with reason")
 
     def parse_Assert(self):
         test_expr = Expr.parse_value_expr(self.stmt.test, self.context)
-
         if self.stmt.msg:
             return self._assert_reason(test_expr, self.stmt.msg)
         else:
             return IRnode.from_list(["assert", test_expr], error_msg="user assert")
 
     def parse_Raise(self):
         if self.stmt.exc:
+            if isinstance(self.stmt.exc, vy_ast.Name) and self.stmt.exc.id == "UNREACHABLE":
+                self.stmt.exc.id = "raise UNREACHABLE"
```

Comment:
why modify the ast here?

Reply:
as a way to tag it as raise unreachable so that it doesn't get passed into IRnode.from_list(["assert_unreachable", test_expr]); otherwise by the time you get to _assert_reason there's no way that I know of to distinguish it

Reply:
that's why the test expr is passed in as None, so `assert_reason` can distinguish -- see the final branch in the `_assert_reason()` implementation.